cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowConfigurer.java

```diff
@@ -43,14 +48,17 @@
   private String name;
   private String description;
   private Map<String, String> properties;
+  private final Map<String, DatasetCreationSpec> localDatasetSpecs = new HashMap<>();
   private int nodeIdentifier = 0;
+  private final DefaultAppConfigurer appConfigurer;
 
   private final List<WorkflowNode> nodes = Lists.newArrayList();
 
-  public DefaultWorkflowConfigurer(Workflow workflow) {
+  public DefaultWorkflowConfigurer(Workflow workflow, DefaultAppConfigurer appConfigurer) {
```

Comment:
Do you need methods only defined on `DefaultAppConfigurer`? Or all you need is a `DatasetConfigurer`?